### ISSUE TITLE
Buggy handling of interrupted map downloads

### ISSUE THREAD

**Issue by pastk** [organicmaps/organicmaps#2436]
- [ ] Map size for interrupted download is always shown as "1 MB" (expected: actual map size to download)
- [ ] If there is a single interrupted map inside of a big country, then a tap on the "!" (error) country icon tries to download a whole country (or at least free space estimation is done for the whole country), expected: resume interrupted maps only inside this country.
- [ ] If interrupted download is cancelled by the user then temp files are not deleted.
- [ ] If there was an interrupted update then the map will look like not downloaded at all and the "Update All" button changes to "Download All"

Details and steps to reproduce.
(to emulated download interruption turn off cellular data in android settings while download is in progress, after a bit of time an error will be shown and map's icon will change to the red "!")

"1 MB" and temp files case:
1. Start map downloading
2. Interrupt
3. Map's size will change to "1 MB" and the icon will change to a red "!"
4. Long tap on the map and choose "cancel"
5. The map will disappear from the downloader screen
6. Check map files folder, there will be corresponding `*.mwm.ready.downloading` and `*.mwm.ready.resume` temp files

Wrong free space estimation case:
1. Start a small single map download from a big country (e.g. Russia), ensure you don't have enough storage space for the whole country.
2. Interrupt
4. Go back to the list of downloaded maps
5. The whole country will have a red "!" icon and there will be a "Download All" button in the bottom
6. Tap on country's icon
7. An "insufficient space" error will be shown, as like the downloader is trying to download a whole country
8. Go inside of the country again (tap on country name)
9. Tap on interrupted map icon
10. The download will resume (free space requirement is estimated properly)

![interrupted](https://user-images.githubusercontent.com/18434508/164503004-e6eacd4d-af3c-40ff-a2e2-b5e76b95136a.png)

Updates case:
1. Start maps updating via "Update All"
2. Interrupt
3. All updates will get a red "!" icon, a button in the bottom will change to "Download All"
4. Resume update of any interrupted country
5. Cancel it by tapping a cross icon
6. All red "!" icons of updatable maps will change back to yellow "update" icons

Everything is tested on Android 5.0 emulator with maps residing on SD card.

**Comment by vrubleg** [comment]
1. Turn on airplane mode.
2. Start to download a map.
3. Remove this not yet downloaded map.
4. Restart Organic Maps (force close and start again).

The result: Organic Maps shows the removed map and tries to download it again.

**Comment by biodranik** [comment]
Wow, the download is not canceled properly. CC @vng

**Comment by jufelka** [comment]
Are there any solution to this bug? Can't download any map

**Comment by biodranik** [comment]
@jufelka There are many issues described above. What is your exact case? Do you use Android? What did you do and what do you observe?

**Comment by jufelka** [comment]
> [@jufelka](https://github.com/jufelka) There are many issues described above. What is your exact case? Do you use Android? What did you do and what do you observe?

I have Android. Trying to download any map (some maps I had downloaded before, and they worked fine) and getting error
"Download has failed. Please make sure your device is connected to the internet". And if I go to the "download maps" section, there will be this failed map file with 1 MB size. I tried to cancel it and download again, but every time I have the same error with different maps.

**Comment by biodranik** [comment]
@jufelka
1. What is your Organic Maps app version? Is it recent?
2. Where did you install it from?
3. Did downloading stop working completely at some moment? When?
4. Which region/country are you in? Could it be that our servers are blocked by providers/government for you?

**Comment by jufelka** [comment]
> [@jufelka](https://github.com/jufelka)
> 
> 1. What is your Organic Maps app version? Is it recent?
> 2. Where did you install it from?
> 3. Did downloading stop working completely at some moment? When?
> 4. Which region/country are you in? Could it be that our servers are blocked by providers/government for you?

I see the problem. Based on the google play info my version is "Last updated Oct 23, 2025", but in the app I see "2023.08.18-8-Google". I thought it should update automatically. So I've just installed a new version from here. But as I understand there will be no automatically update in this case?


I was

**Comment by biodranik** [comment]
You have 2023 (two years old) Organic Maps app version, that's why old maps are not downloaded anymore. Please update it.

**Comment by jufelka** [comment]
> You have 2023 (two years old) Organic Maps app version, that's why old maps are not downloaded anymore. Please update it.

Yeah, I did it. Now all is fine. I was misled by google play message, that I have actual version, because it showed me "Last updated Oct 23, 2025"
